docs/RFCS/typing.md

```diff
@@ -552,6 +552,33 @@ being type checked, it is up to the caller to assert a specific type
 is returned from the expression typing and throw a type checking error
 if necessary.
 
+A desired type will fit into one of three classes: fully-specified types,
+partially-specified types, or unspecified (wildcard) types. Fully-specified types
+are types where the base type and all parameterized types (if applicable) are SQL
+types. Partially-specified types are types where the base type is a SQL type, but 
+one or more parameterized types is a wildcard. Unspecified types are where the base 
+class itself is a wildcard. The table below gives examples of each.
+
+| desired type class     | examples
```

Comment:
The classification here is a bit cumbersome to understand and to fit into the description below. Try this: 

"""
Desired type classes can be:

- fully unspecified (the top level specification is a wildcard)
- structurally specified (the top level specification is not a wildcard, but the desired type may contain wildcards)

We also say that a desired type is "fully specified" if it doesn't contain any wildcard.
"""

Also I'd say the qualifier "specified" only applies to desired type *patterns*, not to *types* themselves. 
The name for a *type* that has no open parameters any more is *"monomorphic"*.